protocols/dcutr/src/behaviour_impl.rs

```diff
@@ -237,11 +239,82 @@ impl Behaviour {
 }
 
 impl NetworkBehaviour for Behaviour {
-    type ConnectionHandler = handler::Prototype;
+    type ConnectionHandler = Either<
+        handler::relayed::Handler,
+        Either<handler::direct::Handler, dummy::ConnectionHandler>,
+    >;
     type OutEvent = Event;
 
-    fn new_handler(&mut self) -> Self::ConnectionHandler {
-        handler::Prototype
+    fn handle_established_inbound_connection(
+        &mut self,
+        _connection_id: ConnectionId,
```

Comment:
Why the `_`? It is used further below in the `remove` call.

Reply:
I opened https://github.com/libp2p/rust-libp2p/pull/3484 to prevent this in the future.